libconnman-qt/networkmanager.cpp

```diff
@@ -827,62 +830,87 @@ void NetworkManager::propertyChanged(const QString &name, const QDBusVariant &va
     propertyChanged(name, value.variant());
 }
 
-void NetworkManager::updateDefaultRoute()
+NetworkService* NetworkManager::selectDefaultRoute(const QString &path)
 {
-    QString defaultNetDev;
-    QFile routeFile("/proc/net/route");
-    if (routeFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
-        QTextStream in(&routeFile);
-        QString line = in.readLine();
-        while (!line.isNull()) {
-            QStringList lineList = line.split('\t');
-            if (lineList.size() >= 11) {
-                if ((lineList.at(1) == "00000000" && lineList.at(3) == "0003") ||
-                   (lineList.at(0).startsWith("ppp") && lineList.at(3) == "0001")) {
-                    defaultNetDev = lineList.at(0);
-                    break;
-                }
+    NetworkService *newDefaultRoute;
```

Comment:
As far as I see it looks that all code branches that returns this assigns something to this. That said, for future it'd be safer to initialize this here to nullptr.

Reply:
Initialized in d90b034ef0128e0196b1ca097354074b1e695b79 to nullptr.